src/notebooks/.ipynb_checkpoints/ELR Profiling Template-checkpoint.ipynb

```diff
@@ -0,0 +1,280 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# ELR Data Profiling Template"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This notebook is a template for how we might use some basic python data analysis (i.e. pandas) to profile incoming ELR data. The code is likely adaptable to ECR data as well, if it fits into a pipe-delimited HL7 v2.XX format. This notebook will attempt to answer extremely basic questions involving the cleanliness and formatting of the data, such as proportion of missing fields per field, proportion of missing required fields, proportion of records with demographic info, etc."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "Note on use: the HL7.Container object parsed by the hl7 package\n",
+    "is a subclass of python list, which gives us easy access for indices:\n",
+    "  Message\n",
+    "  Segment\n",
+    "  Field\n",
+    "  Repetition\n",
+    "  Component\n",
+    "'''\n",
+    "import hl7\n",
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Location of file to open\n",
+    "import os\n",
+    "\n",
+    "HOME_DIR = os.path.expanduser('~')\n",
+    "DATA_DIR = os.path.join(HOME_DIR, 'inventory', 'SampleData', 'PRE') # THIS IS THE ONLY HARD CODE PATH YOU NEED TO CHANGE\n",
+    "IN_FILE = os.path.join(DATA_DIR, 'PreTranslation_to_State_VA.hl7')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "'''\n",
+    "May need to write a different type of file parser depending on format\n",
+    "of real file. The python HL7 library doesn't read from files, just individual\n",
+    "messages, so we have to concatenate them in a meaningful format. Parsing\n",
+    "via string concatenation is suggested in their docs, since newlines and \n",
+    "carriage returns delimit segments and blank lines delimit messages themselves.\n",
+    "'''\n",
+    "msgs = []\n",
+    "with open(IN_FILE, 'r') as fp:\n",
+    "    msg = ''\n",
+    "    for line in fp:\n",
+    "        if line.strip() == '':\n",
+    "            msgs.append(msg)\n",
+    "            msg = ''\n",
+    "        else:\n",
+    "            strp_line = line.strip()\n",
+    "            msg += '\\r' + strp_line\n",
+    "\n",
+    "# Verify correctness of data loading: I count ten messages in this \n",
+    "# representative set; all messages MUST begin with the message\n",
+    "# header signifier 'MSH'\n",
+    "expected_msgs = 10\n",
+    "assert len(msgs) == expected_msgs\n",
+    "starts_with_msg_header = [x.split('|')[0].strip() == 'MSH' for x in msgs]\n",
+    "assert all(starts_with_msg_header)\n",
```

Comment:
I see, so this assertion is hardcoded to 10 based on that specific file right? So if we were to run this in production on real Virginia data we should take this out?

Reply:
Correct! In prod, we could a) change this to an assertion as to the length of whatever snapshot we're looking at, or b) eliminate this and just stick with the sanitation check that all messages start with `MSH`

Reply:
Great!